src/parser/mod.rs

```diff
@@ -367,14 +367,14 @@ fn precedence(token: &Token) -> Option<(i8, bool)> {
     match token {
         Token::Semicolon => Some((0, false)),
         Token::Comma => Some((1, true)),
-        Token::ApplyLeft => Some((2, true)),
-        Token::ApplyRight => Some((3, false)),
-        Token::Or => Some((4, false)),
-        Token::And => Some((5, false)),
-        Token::EqualEqual | Token::Is | Token::Isnt | Token::NotEqual | Token::GreaterThan | Token::LessThan | Token::GreaterThanOrEqual | Token::LessThanOrEqual => Some((7, false)),
-        Token::In => Some((8, false)),
-        Token::Append => Some((9, false)),
-        Token::Range => Some((10, false)),
+        Token::Or => Some((2, false)),
+        Token::And => Some((3, false)),
+        Token::EqualEqual | Token::Is | Token::Isnt | Token::NotEqual | Token::GreaterThan | Token::LessThan | Token::GreaterThanOrEqual | Token::LessThanOrEqual => Some((5, false)),
+        Token::In => Some((6, false)),
+        Token::Append => Some((7, false)),
+        Token::Range => Some((8, false)),
+        Token::ApplyRight => Some((9, false)),
+        Token::ApplyLeft => Some((10, true)),
```

Comment:
I'm rethinking this - I think ApplyLeft/Right should actually be pretty low in precedence. Possibly lower than Token::Comma.
Would you expect `1, 2 |> print` to print out `1, 2` or just the `2`?

Looking at OCaml for inspiration, OCaml actually has `|>` just above Token::And. That'd allow for `true and false |> not` which seems a bit odd to me.

I think I'm leaning towards it being below Token::Comma with `1, 2 |> print` printing out `1, 2`.

Reply:
I would be more used to the `1, 2 |> print` printing `2` than `1, 2`. I believe that would be the case in OCaml as well.
I haven't used piping in tuples that much, though, so I don't have a strong opinion about it

Reply:
I did a quick survey on the discord, looks like most expect it to output `1, 2`. I think lets just go with that, if it ends up being confusing it is easy to change later